<a href="https://colab.research.google.com/github/abbyWoot/AML/blob/master/AML_Project4_JiahuiWu.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

##**Advanced Machine Learning Project4**

@student: Jiahui Wu Abby

In [0]:
#Set Up. 
# To support both python 2 and python 3
from __future__ import division, print_function, unicode_literals

# Scikit-Learn ≥0.20 is required
import sklearn
assert sklearn.__version__ >= "0.20"

# Common imports
import numpy as np
import os
import tarfile
import urllib
import csv

# to make this notebook's output stable across runs
np.random.seed(42)

# To plot pretty figures
%matplotlib inline 
import matplotlib as mpl
import matplotlib.pyplot as plt
mpl.rc('axes', labelsize=14)
mpl.rc('xtick', labelsize=12)
mpl.rc('ytick', labelsize=12)

# Where to save the figures
PROJECT_ROOT_DIR = "."
CHAPTER_ID = "unsupervised_learning"

def save_fig(fig_id, tight_layout=True):
    path = os.path.join(PROJECT_ROOT_DIR, "images", CHAPTER_ID, fig_id + ".png")
    print("Saving figure", fig_id)
    if tight_layout:
        plt.tight_layout()
    plt.savefig(path, format='png', dpi=300)

# Ignore useless warnings (see SciPy issue #5998)
import warnings
warnings.filterwarnings(action="ignore", message="^internal gelsd")

In [0]:
#load data using Pandas
DOWNLOAD_ROOT = "https://archive.ics.uci.edu/ml/machine-learning-databases/"
PM25_PATH = os.path.join("00381", "PRSA_data_2010.1.1-2014.12.31")
PM25_URL = DOWNLOAD_ROOT + "00381/PRSA_data_2010.1.1-2014.12.31.csv"


In [0]:
import pandas as pd

def load_pm25_data(pm25_path=PM25_PATH):
    csv_path = os.path.join(pm25_path, "PRSA_data_2010.1.1-2014.12.31.csv")
    return pd.read_csv(csv_path)

Now let's take a look at the top five rows using the DataFrames' head() method

In [19]:
pm25 = load_pm25_data()
pm25.head()

No  year  month  day  hour  pm2.5  DEWP  TEMP    PRES cbwd    Iws  Is  Ir
0   1  2010      1    1     0    NaN   -21 -11.0  1021.0   NW   1.79   0   0
1   2  2010      1    1     1    NaN   -21 -12.0  1020.0   NW   4.92   0   0
2   3  2010      1    1     2    NaN   -21 -11.0  1019.0   NW   6.71   0   0
3   4  2010      1    1     3    NaN   -21 -14.0  1019.0   NW   9.84   0   0
4   5  2010      1    1     4    NaN   -20 -12.0  1018.0   NW  12.97   0   0

There are 13 attributes: No, year, month, day, hour, pm2.5, DEWP, TEMP, PRES, cbwd, Iws, Is, Ir

In [20]:
pm25.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 43824 entries, 0 to 43823
Data columns (total 13 columns):
No       43824 non-null int64
year     43824 non-null int64
month    43824 non-null int64
day      43824 non-null int64
hour     43824 non-null int64
pm2.5    41757 non-null float64
DEWP     43824 non-null int64
TEMP     43824 non-null float64
PRES     43824 non-null float64
cbwd     43824 non-null object
Iws      43824 non-null float64
Is       43824 non-null int64
Ir       43824 non-null int64
dtypes: float64(4), int64(8), object(1)
memory usage: 4.3+ MB


Above info(0 method showed us that there are 43,824 instances in the dataset, while the pm2.5 attribute has only 41,757 non-null values, meaning that 2067 cases are missing this feature. We will need to take care of this later. 

All attributes are numberical, except the cbwd field. Its type is object, so it could hold any kind of Python object. 

In [21]:
pm25.describe()

No          year  ...            Is            Ir
count  43824.000000  43824.000000  ...  43824.000000  43824.000000
mean   21912.500000   2012.000000  ...      0.052734      0.194916
std    12651.043435      1.413842  ...      0.760375      1.415867
min        1.000000   2010.000000  ...      0.000000      0.000000
25%    10956.750000   2011.000000  ...      0.000000      0.000000
50%    21912.500000   2012.000000  ...      0.000000      0.000000
75%    32868.250000   2013.000000  ...      0.000000      0.000000
max    43824.000000   2014.000000  ...     27.000000     36.000000

[8 rows x 12 columns]